### PR TITLE
Add dockerfiles for testing before release

### PR DESCRIPTION
**Pull request by kunaltyagi**
* Adds a dockerfile for building PCL on `debian:testing`
* Adds a dockerfile for building PCL and `perception_pcl` on `kinetic` and `melodic` flavors of ROS
* Adds a CI to perform some tasks of the release with a manual trigger
* Kinetic `perception_pcl` is broken since 1.10, and can't be targeted by PCL
* Tested manually for melodic

Offshoots of this PR:
* PR ros-perception/perception_pcl#273 to address changes in downstream project to handle change of pointers

TODO after this PR:
* Test CI

### CODE REVIEW

**Comment by kunaltyagi** [comment]
Need help testing the dockerfiles

**Comment by jspricke** [comment]
Note that perception_pcl is in Debian (testing), so you could go with apt source -b ros-perception-pcl

File: .dev/docker/release/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,42 @@
+FROM debian:latest
```
Comment (kunaltyagi) [inline .dev/docker/release/Dockerfile:1]:
```suggestion
FROM debian:testing
```

**Comment by kunaltyagi** [comment]
Azure is running out of disk space for debian testing

**Comment by kunaltyagi** [comment]
PR opened to make sure downstream target `pcl_ros` works after the smart pointer API break.

All images have been verified manually (not on CI) before the API break

**Comment by kunaltyagi** [comment]
Should I remove the commented code? The plan was to test it in this PR, but my account's CI doesn't have enough space to build PCL

**Comment by taketwo** [comment]
> could you please contact Azure to increase CI space just a bit more so Windows doesn't OOM out randomly

They said they have to technical possibility to increase the space.

**Comment by kunaltyagi** [comment]
> They said they have to

Did you mean *"They said they have **no**"* instead?

**Comment by taketwo** [comment]
Yep, **no** technical possibility, sorry. To quote their answer:

> We do not have the ability to increase the free disk space limit on hosted machines. Everyone in free tier and paid tier get the same free disk space.

**Comment by kunaltyagi** [comment]
Got it. In which case, I guess we see what happens after this gets reviewed, merged and a new pipeline is created.

File: .ci/azure-pipelines/release.yaml [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    - checkout: self
+      # find the commit hash on a quick non-forced update too
+      fetchDepth: 10
+    - task: Bash@3
+      displayName: "Remove git files"
+      inputs:
+        targetType: 'inline'
+        script: 'rm -fr ./.git'
+        workingDirectory: '$(Build.SourcesDirectory)'
+        failOnStderr: true
```
Comment (SergioRAgostinho) [inline .ci/azure-pipelines/release.yaml:94]: You can pass `--exclude-vcs` to tar and eliminate this step.

Comment (kunaltyagi) [reply .ci/azure-pipelines/release.yaml:94]: AFAIK, option is not available: https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema?view=azure-devops&tabs=schema%2Cparameter-schema#checkout

Alternative is to `checkout: none` and use a raw command

Comment (SergioRAgostinho) [reply .ci/azure-pipelines/release.yaml:94]: Ignore

**Comment by kunaltyagi** [comment]
Yes. That's sadly needed because the GitHub interface isn't quite the same as the GitHub task on Azure CI. I tried with not uploading a source, but then the release didn't have anything

**Comment by kunaltyagi** [comment]
Merging to start testing with PCL's pipeline instead of mine

**Pull request merged**